src/v/cluster/controller_backend.cc

```diff
@@ -953,24 +954,38 @@ ss::future<> controller_backend::delete_non_replicable_partition(
     vlog(clusterlog.trace, "removing {} from shard table at {}", ntp, rev);
     co_await _shard_table.invoke_on_all(
       [ntp, rev](shard_table& st) { st.erase(ntp, rev); });
-    auto log = _storage.local().log_mgr().get(ntp);
-    if (log && log->config().get_revision() < rev) {
-        co_await _storage.local().log_mgr().remove(ntp);
+    auto nr_partition = _nr_partition_manager.local().get(ntp);
+    if (nr_partition && nr_partition->get_revision_id() < rev) {
+        co_await _nr_partition_manager.local().remove(ntp);
     }
 }
 
 ss::future<std::error_code> controller_backend::create_non_replicable_partition(
   model::ntp ntp, model::revision_id rev) {
-    auto cfg = _topics.local().get_topic_cfg(model::topic_namespace_view(ntp));
-    if (!cfg) {
+    auto& topics_map = _topics.local().topics_map();
+    auto tt_md = topics_map.find(model::topic_namespace_view(ntp));
+    if (tt_md == topics_map.end()) {
         // partition was already removed, do nothing
         co_return errc::success;
     }
     vassert(
-      !_storage.local().log_mgr().get(ntp),
-      "Log exists for missing entry in topics_table");
-    auto ntp_cfg = cfg->make_ntp_config(_data_directory, ntp.tp.partition, rev);
-    co_await _storage.local().log_mgr().manage(std::move(ntp_cfg));
+      !tt_md->second.is_topic_replicable(),
+      "Replicable topic reached non-replicable API");
+    auto nr_partition = _nr_partition_manager.local().get(ntp);
+    if (likely(!nr_partition)) {
+        // get_source_topic will assert if incorrect API is used
+        auto src_partition = _partition_manager.local().get(model::ntp(
+          ntp.ns, tt_md->second.get_source_topic(), ntp.tp.partition));
+        if (!src_partition) {
+            co_return errc::partition_not_exists;
+        }
+        auto ntp_cfg = tt_md->second.get_configuration().cfg.make_ntp_config(
+          _data_directory, ntp.tp.partition, rev);
+        co_await _nr_partition_manager.local().manage(
+          std::move(ntp_cfg), src_partition);
+    } else if (nr_partition->get_revision_id() < rev) {
```

Comment:
Is `<` correct, or should it be `<=`?

Reply:
https://github.com/vectorizedio/redpanda/blob/dev/src/v/cluster/controller_backend.cc#L1003 Modeled after `create_partition`